frontend/src/metabase/querying/parameters/utils/query.unit.spec.ts

```diff
@@ -0,0 +1,314 @@
+import { createMockMetadata } from "__support__/metadata";
+import * as Lib from "metabase-lib";
+import {
+  columnFinder,
+  createQuery,
+  createQueryWithClauses,
+} from "metabase-lib/test-helpers";
+import type {
+  ParameterTarget,
+  ParameterType,
+  ParameterValueOrArray,
+} from "metabase-types/api";
+import { createMockField } from "metabase-types/api/mocks";
+import {
+  ORDERS_ID,
+  createOrdersTable,
+  createProductsTable,
+  createSampleDatabase,
+} from "metabase-types/api/mocks/presets";
+
+import { applyParameter } from "./query";
+
+type FilterParameterCase = {
+  type: ParameterType;
+  target: ParameterTarget;
+  value: ParameterValueOrArray;
+  expectedDisplayName: string;
+};
+
+type TemporalUnitParameterCase = {
+  value: ParameterValueOrArray;
+  expectedDisplayName: string;
+};
+
+const BOOLEAN_FIELD = createMockField({
+  id: 102,
+  table_id: ORDERS_ID,
+  name: "IS_TRIAL",
+  display_name: "Is trial",
+  base_type: "type/Boolean",
+  effective_type: "type/Boolean",
+  semantic_type: "type/Category",
+});
+
+const ORDERS_TABLE = createOrdersTable();
+
+const METADATA = createMockMetadata({
+  databases: [
+    createSampleDatabase({
+      tables: [
+        createOrdersTable({
+          fields: [...(ORDERS_TABLE.fields ?? []), BOOLEAN_FIELD],
+        }),
+        createProductsTable(),
+      ],
+    }),
+  ],
+});
+
+describe("applyParameter", () => {
+  describe("filter parameters", () => {
+    const query = createQuery({ metadata: METADATA });
+    const stageIndex = 0;
+    const columns = Lib.filterableColumns(query, stageIndex);
+    const findColumn = columnFinder(query, columns);
+
+    function getFilterColumnTarget(
+      tableName: string,
+      columnName: string,
+    ): ParameterTarget {
+      const column = findColumn(tableName, columnName);
+      const columnRef = Lib.legacyRef(query, stageIndex, column);
+      return ["dimension", columnRef] as ParameterTarget;
```

Comment:
We can avoid the cast by introducing a type guard checking `columnRef[0] !== "aggregation"`.
We can reuse it in `temporal unit parameters` tests at the bottom which have the same cast.

```suggestion
      if (isAggregationRef(columnRef)) {
        throw new Error("aggregation is not a valid parameter target");
      }
      
      return ["dimension", columnRef] as ParameterTarget;
```